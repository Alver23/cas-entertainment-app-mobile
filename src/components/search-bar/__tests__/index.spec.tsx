// Dependencies
import React, { Dispatch } from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { SearchBar, Button } from 'react-native-elements';

import Search from '../index';

describe('<Search />', () => {
  let component: ShallowWrapper;

  const mockState: [unknown, Dispatch<Function>] = [null, jest.fn()];
  const props = {
    placeholder: 'placeholder',
    buttonTitle: 'buttonTitle',
    onPress: jest.fn(),
  };

  beforeEach(() => {
    jest.spyOn(React, 'useState').mockReturnValue(mockState);
    component = shallow(<Search {...props} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should call the onChangeText void', () => {
    const onChangeText = component.find(SearchBar).prop('onChangeText') as any;
    const mockText = 'fake test';
    onChangeText(mockText);
    expect(mockState[1]).toHaveBeenCalledWith(mockText);
  });

  it('should call the onPress void', () => {
    const onPress = component.find(Button).prop('onPress') as any;

    onPress();
    expect(props.onPress).toHaveBeenCalled();
  });
});
