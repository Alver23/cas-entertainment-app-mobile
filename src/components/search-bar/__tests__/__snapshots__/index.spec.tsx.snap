// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginHorizontal": 16,
    }
  }
>
  <ForwardRef(Themed.SearchBar)
    containerStyle={
      Object {
        "backgroundColor": "transparent",
        "flex": 4,
      }
    }
    inputContainerStyle={
      Object {
        "backgroundColor": undefined,
      }
    }
    onChangeText={[Function]}
    placeholder="placeholder"
    searchIcon={
      Object {
        "iconStyle": Object {
          "color": "#2089dc",
        },
        "size": 28,
      }
    }
    showCancel={true}
    value={null}
  />
  <ForwardRef(Themed.Button)
    containerStyle={
      Object {
        "flex": 1,
      }
    }
    onPress={[Function]}
    title="buttonTitle"
  />
</View>
`;
